programs/stake_api/src/stake_state.rs

```diff
@@ -326,13 +340,14 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
             return Err(InstructionError::InsufficientFunds);
         }
 
-        if let StakeState::Uninitialized = self.state()? {
+        if let StakeState::Lockup(lockup) = self.state()? {
```

Comment:
stakes should be delegatable if they are locked up, just claiming rewards is locked

Reply:
yes, that's what this says

Reply:
StakeState::Uninitialized is fleeting, we set the stake to lockup at creation time.  the owner of the account can later delegate (and I'm going to also add re-delegation)